openlibrary/core/waitinglist.py

```diff
@@ -111,38 +111,45 @@ def query(cls, **kw):
         return [cls(row) for row in rows]
 
     @classmethod
-    def new(cls, **kw):
+    def new(cls, **kw) -> 'WaitingLoan | None':
         user_key = kw['user_key']
         itemname = kw.get('itemname', '')
         if not itemname:
             account = OpenLibraryAccount.get(key=user_key)
+            if account is None:
+                return None
```

Comment:
To keep this behaviour consistent with how it was before, we must throw an error here to prevent having to check/update the downstream code to handle `itemname` being `None`, or to handle this function returning `None` in a case where previously it would error.

```suggestion
            if account is None:
                raise ValueError('Unable to get Open Library account itemname')
```

(Also much to my dismay, TIL `assert` is discouraged from prod code?! So we have to write out this whole raise -_-)